src/CheckedFile.cpp

```diff
@@ -384,25 +437,38 @@ void CheckedFile::seek(uint64_t offset, OffsetMode omode)
 
 uint64_t CheckedFile::lseek64(int64_t offset, int whence)
 {
+   int64_t result = -1;
+   if(fd_ >= 0){
 #if defined(_WIN32)
 #  if defined(_MSC_VER) || defined(__MINGW32__) //<rs 2010-06-16> mingw _is_ WIN32!
-   __int64 result = _lseeki64(fd_, offset, whence);

```

Comment:
You're changing the return type for this. I would prefer not to. `_lseeki64` returns `__int64`.

If you want to unify these in another PR, maybe there's a better function to use than `_lseeki64` on Windows?